### PR TITLE
[patch] Update git checkout to newer. JB#57049

### PR DESCRIPTION
**Pull request by pvuorela**
Fixing compatibility with bison 3.8.

- Previous patch no longer needed, upstream has proper gnulib usage.
- Needed new patch for execute() api changes in gnulib
- .changes removed, not our history and had bogus date
- Added %license

@mlehtima

### CODE REVIEW

**Review by mlehtima — APPROVED** [review]

File: rpm/patch.spec [hunk 1]
```diff
@@ -2,11 +2,10 @@ Name:       patch
 Summary:    The GNU patch command, for modifying/upgrading files
 Version:    2.7.6
 Release:    1
-Group:      Development/Tools
 License:    GPLv3
-URL:        http://www.gnu.org/software/patch/patch.html
+URL:        https://github.com/sailfishos/patch
 Source0:    ftp://ftp.gnu.org/gnu/patch/patch-%{version}.tar.xz
```
Comment (saukko) [inline rpm/patch.spec]: Extra point from changing this to %{name}-%{version}.tar.xz

Comment (Thaodan) [reply rpm/patch.spec]: zstd might also work

**Review by Thaodan — APPROVED** [review]
LGTM, except Markos comment.

File: rpm/patch.spec (hunk 1 above)
Comment (pvuorela) [reply rpm/patch.spec]: Sure.

**Review by mlehtima — APPROVED** [review]

**Pull request merged**